incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -136,6 +143,31 @@ func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logg
 	return r.updateObjects(ctx, log, nm)
 }
 
+func (r *HierarchyConfigReconciler) onMissingNamespace(log logr.Logger, nm string) {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nm)
+
+	if ns.Exists() {
+		r.enqueueAffected(log, "relative of deleted namespace", ns.RelativesNames()...)
+		ns.UnsetExists()
+		log.Info("Removed namespace")
+	}
+}
+
+func (r *HierarchyConfigReconciler) updateFinalizers(ctx context.Context, log logr.Logger, inst *api.HierarchyConfiguration, nsInst *corev1.Namespace, hnsnms []string) {
+	switch {
+	case nsInst.DeletionTimestamp.IsZero():
+		// If the namespace is not being deleted, just remove finalizers.
```

Comment:
Can you explain a bit better _why_ we're doing this (both comments in this function)?

Reply:
This happens when it's not a cascading deletion and user just deletes the HC instance. We should remove the finalizer so it can be purged. A new default HC instance will then be recreated by a next HC reconciliation.